data/binding/webstring.go

```diff
@@ -59,3 +45,28 @@ func (s *webSocketString) Close() error {
 
 	return s.conn.Close()
 }
+
+func (s *webSocketString) Get() (string, error) {
+	if s.prev != nil {
+		err := s.prev
+		s.prev = nil
+		return "", err
+	}
+
+	return s.String.Get()
+}
+
+func (s *webSocketString) readMessages() {
+	for {
+		_, p, err := s.conn.ReadMessage()
+		if err != nil { // permanent (could be connection closed)
+			s.prev = err
+			return
+		}
+
+		err = s.Set(string(p))
+		if err != nil { // not permanent
```

Comment:
Seems odd that a subsequent Get would return an error from Set - this error would normally be from a later binding, eg converter.

Maybe I'm miss understanding the use cases; 1) dev creates websocket binding and attaches it to a label, websocket fails to connect so dev must call Get to see what the issue was? 2) websocket connects and receives a message, but binding fails to set the string, so again dev must call Get to see what the issue was? Worse is that either error can occur at any time, so the dev would need to poll Get :(

Reply:
I think it might be nicer to be able to set a callback `OnError` so the dev can explicitly handle failures, rather than this implicit behaviour.

Reply:
Sorry I missed that this error can never occur, so I now ignore it with a comment

Reply:
> 1) dev creates websocket binding and attaches it to a label, websocket fails to connect so dev must call Get to see what the issue was?

If it fails to connect the error is returned from the constructor, no need to call into Get - those are for messages happening later.